test/integration/rules/runner.js

```diff
@@ -102,12 +102,27 @@
 							axe.run(
 								fixture,
 								{
+									/**
+									 * The debug flag helps log errors in a fairly detailed fashion,
+									 * when tests fail in webdriver
+									 */
+									debug: true,
 									performanceTimer: false,
 									runOnly: { type: 'rule', values: [ruleId] }
 								},
 								function(err, r) {
+									// assert that there are no errors
 									assert.isNull(err);
+									// assert that result is defined
+									assert.isDefined(r);
+									// assert that result has certain keys
+									assert.hasAnyKeys(r, ['incomplete', 'violations', 'passes']);
+									// flatten results
 									results = flattenResult(r);
+									// assert incomplete in results does not have error
+									if (results.incomplete) {
+										assert.isUndefined(results.incomplete.error);
```

Comment:
results.incomplete should be an array here, no? So shouldn't this be:

```js
results.incomplete.forEach(function (result) {
  assert.isUndefined(results.error)
});
```

Reply:
Well, as a part of flattenResult in L121, we pluck the first item in the incomplete array.

If we were to move the assertion before flattening the results, we have to `forEach`. No harm in either way. Shall refactor.